<!DOCTYPE html>
<!-- START: inst/pkgdown/templates/layout.html --><!-- Generated by pkgdown: do not edit by hand --><html lang="en" data-bs-theme="auto"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><title>Metagenomics Workshop Overview: Summary and Schedule</title><meta name="viewport" content="width=device-width, initial-scale=1"><script src="../assets/themetoggle.js"></script><link rel="stylesheet" type="text/css" href="../assets/styles.css"><script src="../assets/scripts.js" type="text/javascript"></script><!-- mathjax --><script type="text/x-mathjax-config">
    MathJax.Hub.Config({
      config: ["MMLorHTML.js"],
      jax: ["input/TeX","input/MathML","output/HTML-CSS","output/NativeMML", "output/PreviewHTML"],
      extensions: ["tex2jax.js","mml2jax.js","MathMenu.js","MathZoom.js", "fast-preview.js", "AssistiveMML.js", "a11y/accessibility-menu.js"],
      TeX: {
        extensions: ["AMSmath.js","AMSsymbols.js","noErrors.js","noUndefined.js"]
      },
      tex2jax: {
        inlineMath: [['\\(', '\\)']],
        displayMath: [ ['$$','$$'], ['\\[', '\\]'] ],
        processEscapes: true
      }
    });
    </script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><!-- Responsive Favicon for The Carpentries --><link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="manifest" href="../site.webmanifest"><link rel="mask-icon" href="../safari-pinned-tab.svg" color="#5bbad5"><meta name="msapplication-TileColor" content="#da532c"><meta name="theme-color" media="(prefers-color-scheme: light)" content="white"><meta name="theme-color" media="(prefers-color-scheme: dark)" content="black"></head><body>
    <header id="top" class="navbar navbar-expand-md top-nav lab"><svg xmlns="http://www.w3.org/2000/svg" class="d-none"><symbol id="check2" viewbox="0 0 16 16"><path d="M13.854 3.646a.5.5 0 0 1 0 .708l-7 7a.5.5 0 0 1-.708 0l-3.5-3.5a.5.5 0 1 1 .708-.708L6.5 10.293l6.646-6.647a.5.5 0 0 1 .708 0z"></path></symbol><symbol id="circle-half" viewbox="0 0 16 16"><path d="M8 15A7 7 0 1 0 8 1v14zm0 1A8 8 0 1 1 8 0a8 8 0 0 1 0 16z"></path></symbol><symbol id="moon-stars-fill" viewbox="0 0 16 16"><path d="M6 .278a.768.768 0 0 1 .08.858 7.208 7.208 0 0 0-.878 3.46c0 4.021 3.278 7.277 7.318 7.277.527 0 1.04-.055 1.533-.16a.787.787 0 0 1 .81.316.733.733 0 0 1-.031.893A8.349 8.349 0 0 1 8.344 16C3.734 16 0 12.286 0 7.71 0 4.266 2.114 1.312 5.124.06A.752.752 0 0 1 6 .278z"></path><path d="M10.794 3.148a.217.217 0 0 1 .412 0l.387 1.162c.173.518.579.924 1.097 1.097l1.162.387a.217.217 0 0 1 0 .412l-1.162.387a1.734 1.734 0 0 0-1.097 1.097l-.387 1.162a.217.217 0 0 1-.412 0l-.387-1.162A1.734 1.734 0 0 0 9.31 6.593l-1.162-.387a.217.217 0 0 1 0-.412l1.162-.387a1.734 1.734 0 0 0 1.097-1.097l.387-1.162zM13.863.099a.145.145 0 0 1 .274 0l.258.774c.115.346.386.617.732.732l.774.258a.145.145 0 0 1 0 .274l-.774.258a1.156 1.156 0 0 0-.732.732l-.258.774a.145.145 0 0 1-.274 0l-.258-.774a1.156 1.156 0 0 0-.732-.732l-.774-.258a.145.145 0 0 1 0-.274l.774-.258c.346-.115.617-.386.732-.732L13.863.1z"></path></symbol><symbol id="sun-fill" viewbox="0 0 16 16"><path d="M8 12a4 4 0 1 0 0-8 4 4 0 0 0 0 8zM8 0a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 0zm0 13a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 13zm8-5a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2a.5.5 0 0 1 .5.5zM3 8a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2A.5.5 0 0 1 3 8zm10.657-5.657a.5.5 0 0 1 0 .707l-1.414 1.415a.5.5 0 1 1-.707-.708l1.414-1.414a.5.5 0 0 1 .707 0zm-9.193 9.193a.5.5 0 0 1 0 .707L3.05 13.657a.5.5 0 0 1-.707-.707l1.414-1.414a.5.5 0 0 1 .707 0zm9.193 2.121a.5.5 0 0 1-.707 0l-1.414-1.414a.5.5 0 0 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .707zM4.464 4.465a.5.5 0 0 1-.707 0L2.343 3.05a.5.5 0 1 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .708z"></path></symbol></svg><a class="visually-hidden-focusable skip-link" href="#main-content">Skip to main content</a>
  <div class="container-fluid top-nav-container">
    <div class="col-md-8">
      <div class="large-logo">
        <img id="lab-logo" alt="Carpentries Lab" src="../assets/images/lab-logo.svg"><abbr class="badge peer-reviewed" title="This lesson has passed peer review.">
          <a href="https://doi.org/10.21105/jose.00209" class="external-link alert-link">
            <i aria-hidden="true" class="icon" data-feather="user-check" style="border-radius: 5px"></i>
            DOI: 10.21105/jose.00209
          </a>
        </abbr>

      </div>
    </div>
    <div class="selector-container">
      <div id="theme-selector">
        <li class="nav-item dropdown" id="theme-button-list">
          <button class="btn btn-link nav-link px-0 px-lg-2 dropdown-toggle d-flex align-items-center" id="bd-theme" type="button" aria-expanded="false" data-bs-toggle="dropdown" data-bs-display="static" aria-label="Toggle theme (auto)">
            <svg class="bi my-1 theme-icon-active"><use href="#circle-half"></use></svg><i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="bd-theme-text"><li>
              <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="light" aria-pressed="false">
                <svg class="bi me-2 theme-icon"><use href="#sun-fill"></use></svg>
                Light
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
            <li>
              <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="dark" aria-pressed="false">
                <svg class="bi me-2 theme-icon"><use href="#moon-stars-fill"></use></svg>
                Dark
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
            <li>
              <button type="button" class="btn dropdown-item d-flex align-items-center active" data-bs-theme-value="auto" aria-pressed="true">
                <svg class="bi me-2 theme-icon"><use href="#circle-half"></use></svg>
                Auto
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
          </ul></li>
      </div>

      <div class="dropdown" id="instructor-dropdown">
        <button class="btn btn-secondary dropdown-toggle bordered-button" type="button" id="dropdownMenu1" data-bs-toggle="dropdown" aria-expanded="false">
          <i aria-hidden="true" class="icon" data-feather="eye"></i> Instructor View <i data-feather="chevron-down"></i>
        </button>
        <ul class="dropdown-menu" aria-labelledby="dropdownMenu1"><li><button class="dropdown-item" type="button" onclick="window.location.href='../index.html';">Learner View</button></li>
        </ul></div>
    </div>
  </div>
  <hr></header><nav class="navbar navbar-expand-xl bottom-nav lab" aria-label="Main Navigation"><div class="container-fluid nav-container">
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle Navigation">
      <span class="navbar-toggler-icon"></span>
      <span class="menu-title">Menu</span>
    </button>
    <div class="nav-logo">
      <img class="small-logo" alt="Carpentries Lab" src="../assets/images/lab-logo-sm.svg"></div>
    <div class="lesson-title-md">
      Metagenomics Workshop Overview
    </div>
    <div class="search-icon-sm">
      <!-- TODO: do not show until we have search
        <i role="img" aria-label="Search the All In One page" data-feather="search"></i>
      -->
    </div>
    <div class="desktop-nav">
      <ul class="navbar-nav me-auto mb-2 mb-lg-0"><li class="nav-item">
          <span class="lesson-title">
            Metagenomics Workshop Overview
          </span>
        </li>
        <li>
          <a class="nav-link" href="index.html#setup">Setup</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/instructor-notes.html">Instructor Notes</a>
        </li>
        <li class="nav-item dropdown">
          <button class="nav-link dropdown-toggle" id="navbarDropdown" data-bs-toggle="dropdown" aria-expanded="false">
            More <i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu" aria-labelledby="navbarDropdown"><li><a class="dropdown-item" href="AMI-setup.html">Launching your own AMI instances</a></li><li><a class="dropdown-item" href="data.html">Data</a></li><li><a class="dropdown-item" href="faq.html">Frequently Asked Questions</a></li><hr><li><a class="dropdown-item" href="reference.html">Reference</a></li>
          </ul></li>
      </ul></div>
    <!--
    <form class="d-flex col-md-2 search-form">
      <fieldset disabled>
      <input class="form-control me-2 searchbox" type="search" placeholder="" aria-label="">
        <button class="btn btn-outline-success tablet-search-button"  type="submit">
          <i class="search-icon" data-feather="search" role="img" aria-label="Search the All In One page"></i>
        </button>
      </fieldset>
    </form>
    -->
  </div><!--/div.container-fluid -->
</nav><div class="col-md-12 mobile-title">
  Metagenomics Workshop Overview
</div>


    <div class="container">
      <div class="row">
        <!-- START: inst/pkgdown/templates/navbar.html -->
<div id="sidebar-col" class="col-lg-4 overview-sidebar">
  <div id="sidebar" class="sidebar">
      <nav aria-labelledby="flush-headingEleven"><button role="button" aria-label="close menu" alt="close menu" aria-expanded="true" aria-controls="sidebar" class="collapse-toggle" data-collapse="Collapse " data-episodes="Episodes ">
          <i class="search-icon" data-feather="x" role="img"></i>
        </button>
        <div class="sidebar-inner">
          <div class="row mobile-row" id="theme-row-mobile">
            <div class="col" id="theme-selector">
              <li class="nav-item dropdown" id="theme-button-list">
                <button class="btn btn-link nav-link px-0 px-lg-2 dropdown-toggle d-flex align-items-center" id="bd-theme" type="button" aria-expanded="false" data-bs-toggle="dropdown" data-bs-display="static" aria-label="Toggle theme (auto)">
                  <svg class="bi my-1 theme-icon-active"><use href="#circle-half"></use></svg><span class="d-lg-none ms-1" id="bd-theme-text">Toggle Theme</span>
                </button>
                <ul class="dropdown-menu dropdown-menu-right" aria-labelledby="bd-theme-text"><li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="light" aria-pressed="false">
                      <svg class="bi me-2 theme-icon"><use href="#sun-fill"></use></svg>
                      Light
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                  <li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="dark" aria-pressed="false">
                      <svg class="bi me-2 theme-icon"><use href="#moon-stars-fill"></use></svg>
                      Dark
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                  <li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center active" data-bs-theme-value="auto" aria-pressed="true">
                      <svg class="bi me-2 theme-icon"><use href="#circle-half"></use></svg>
                      Auto
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                </ul></li>
            </div>
          </div>
          <div class="row mobile-row">
            <div class="col">
              <div class="sidenav-view-selector">
                <div class="accordion accordion-flush" id="accordionFlush9">
                  <div class="accordion-item">
                    <h2 class="accordion-header" id="flush-headingNine">
                      <button class="accordion-button collapsed" id="instructor" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseNine" aria-expanded="false" aria-controls="flush-collapseNine">
                        <i id="eye" aria-hidden="true" class="icon" data-feather="eye"></i> Instructor View
                      </button>
                    </h2>
                    <div id="flush-collapseNine" class="accordion-collapse collapse" aria-labelledby="flush-headingNine" data-bs-parent="#accordionFlush2">
                      <div class="accordion-body">
                        <a href="../index.html">Learner View</a>
                      </div>
                    </div>
                  </div><!--/div.accordion-item-->
                </div><!--/div.accordion-flush-->
              </div><!--div.sidenav-view-selector -->
            </div><!--/div.col -->

            <hr></div><!--/div.mobile-row -->

          <div class="accordion accordion-flush" id="accordionFlush11">
            <a href="index.html">Home</a>

            <hr class="half-width"><div class="accordion accordion-flush resources" id="accordionFlush12">
              <div class="accordion-item">
                <h2 class="accordion-header" id="flush-headingTwelve">
                  <button class="accordion-button show" id="resources" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseTwelve" aria-expanded="false" aria-controls="flush-collapseTwelve">
                    RESOURCES
                  </button>
                </h2>
                <div id="flush-collapseTwelve" class="accordion-collapse show collapse" aria-labelledby="flush-headingTwelve" data-bs-parent="#accordionFlush12">
                  <div class="accordion-body">
                    <ul><li>
                        <a href="index.html#setup">Setup</a>
                      </li>
                      <li>
                        <a href="../instructor/instructor-notes.html">Instructor Notes</a>
                      </li>
                      <li><a href="AMI-setup.html">Launching your own AMI instances</a></li><li><a href="data.html">Data</a></li><li><a href="faq.html">Frequently Asked Questions</a></li><hr><li><a class="dropdown-item" href="reference.html">Reference</a></li>
                    </ul></div>
                </div>
              </div>
            </div>
            <hr class="half-width resources"><hr class="d-none d-sm-block d-md-none"><div class="d-grid gap-1">

            </div>
          </div><!-- /div.accordion -->
        </div><!-- /div.sidebar-inner -->
      </nav></div><!-- /div.sidebar -->
  </div><!-- /div.sidebar-col -->
<!-- END:   inst/pkgdown/templates/navbar.html-->

        <!-- START: inst/pkgdown/templates/content-syllabus.html -->
<div class="col-xl-12 col-lg-12">
    <main id="main-content" class="main-content"><h1>Summary and Schedule</h1>
      <div class="container lesson-content">
        <p>

          <a href="https://github.com/carpentries-lab/metagenomics-workshop/edit/main/index.md" class="external-link">
            Edit this page <i aria-hidden="true" data-feather="edit"></i>
          </a>
        </p>

      

<p>Data Carpentry aims to teach researchers basic concepts, skills, and
tools for working with data so they can get more done quickly and with
less pain. This workshop uses Data Carpentry’s approach to teach data
management and analysis for metagenomics research, including: best
practices for the organization of bioinformatics projects and data, use
of command-line utilities, use of command-line tools to analyze sequence
quality, use of R studio and use of R libraries to compare diversity
between samples, and connecting to and using cloud computing. This
workshop is designed to be taught over two full days of instruction.</p>
<p>Would you be interested in teaching these materials? We have an <a href="https://join.slack.com/t/metagenomicslesson/shared_invite/zt-1tikq2o0k-VMX7ceLfenbbKheZ7XNCRw" class="external-link">Slack
channel</a> were we will be happy to help you!</p>
<div id="frequently-asked-questions" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="frequently-asked-questions" class="callout-inner">
<h3 class="callout-title">Frequently Asked Questions<a class="anchor" aria-label="anchor" href="#frequently-asked-questions"></a>
</h3>
<div class="callout-content">
<p>Read our <a href="https://carpentries-lab.github.io/metagenomics-workshop/faq/index.html" class="external-link">FAQ</a>
to learn more about Data Carpentry’s Metagenomics workshop as an
Instructor or a workshop host.</p>
</div>
</div>
</div>
<div id="getting-started" class="callout prereq">
<div class="callout-square">
<i class="callout-icon" data-feather="check"></i>
</div>
<div id="getting-started" class="callout-inner">
<h3 class="callout-title">Getting Started<a class="anchor" aria-label="anchor" href="#getting-started"></a>
</h3>
<div class="callout-content">
<p>This lesson assumes that learners have no prior experience with the
tools covered in the workshop.</p>
<p>However, learners are expected to have some familiarity with
biological concepts, including the concept of DNA sequencing, nucleotide
abbreviations, genome, microbiome, and taxonomy. Participants should
bring their own laptops and plan to participate actively.</p>
<p>To get started, follow the directions in the <a href="https://carpentries-lab.github.io/metagenomics-workshop/setup.html" class="external-link">Setup</a>
tab to get access to the required software and data for this
workshop.</p>
</div>
</div>
</div>
<div id="data" class="callout prereq">
<div class="callout-square">
<i class="callout-icon" data-feather="check"></i>
</div>
<div id="data" class="callout-inner">
<h3 class="callout-title">Data<a class="anchor" aria-label="anchor" href="#data"></a>
</h3>
<div class="callout-content">
<p>This workshop uses data from the environmental experiment: <a href="https://elifesciences.org/articles/49816" class="external-link">Genomic adaptations in
information processing underpins trophic strategy in a whole-ecosystem
nutrient enrichment experiment</a>, by Jordan G Okie et al. In this
research, authors compared the differences between the microbial
community in its natural, oligotrophic, phosphorus-deficient
environment, a pond from the Cuatro Ciénegas Basin (CCB), and the same
microbial community under a fertilization treatment.</p>
<p>All of the data used in this workshop can be downloaded from <a href="https://doi.org/10.5281/zenodo.7010950" class="external-link"><img src="https://zenodo.org/badge/DOI/10.5281/zenodo.7010950.svg" alt="DOI" class="figure"></a> More information about this data is available on the <a href="https://carpentries-lab.github.io/metagenomics-workshop/data/index.html" class="external-link">Data
page</a>.</p>
</div>
</div>
</div>
<div class="section level1">
<h1 id="workshop-overview">Workshop Overview<a class="anchor" aria-label="anchor" href="#workshop-overview"></a></h1>
<table class="table"><colgroup><col width="3%"><col width="89%"><col width="7%"></colgroup><thead><tr class="header"><th>Lesson</th>
<th>Overview</th>
<th>Estimated time</th>
</tr></thead><tbody><tr class="odd"><td><a href="https://carpentries-incubator.github.io/organization-metagenomics/" class="external-link">Project
Organization and Management</a></td>
<td>Learn how to structure your metadata, organize and document your
metagenomics data and bioinformatics workflow, and access data on the
NCBI sequence read archive (SRA) database.</td>
<td>1:30 hr</td>
</tr><tr class="even"><td><a href="https://carpentries-incubator.github.io/shell-metagenomics/" class="external-link">Introduction
to the Command Line</a></td>
<td>Learn to navigate your file system, create, copy, move, and remove
files and directories, and automate repetitive tasks using scripts and
wildcards.</td>
<td>4:00 hr</td>
</tr><tr class="odd"><td><a href="https://carpentries-incubator.github.io/introduction-to-R-for-metagenomics/" class="external-link">Introduction
to R</a></td>
<td>Use R studio to manage several data types and structures.</td>
<td>1:00 hr</td>
</tr><tr class="even"><td><a href="https://carpentries-incubator.github.io/metagenomics/" class="external-link">Data
Processing and Visualization for Metagenomics</a></td>
<td>Use command-line tools to perform quality control, metagenomic
assembly, metagenomic binning, taxonomic assignment, and diversity
exploration.</td>
<td>6:30 hr</td>
</tr></tbody></table><!--
# Optional Additional Lessons

| Lesson | Overview |
| ------- | -------- |
| [16S genomics](https://datacarpentry.org/genomics-r-intro/) | Use R to analyze and visualize between-sample variation. |
!--></div>
<div class="section level1">
<h1 id="lessons-reference">Lessons Reference<a class="anchor" aria-label="anchor" href="#lessons-reference"></a></h1>
<p>The content of this page and three of the lessons presented in this
workshop are adapted from lessons on the <a href="https://datacarpentry.org/genomics-workshop/" class="external-link">Data Carpentry
Genomics Workshop</a>.</p>
</div>
<div class="section level1">
<h1 id="teaching-platform">Teaching Platform<a class="anchor" aria-label="anchor" href="#teaching-platform"></a></h1>
<p>This workshop is designed to be run on pre-imaged Amazon Web Services
(AWS) instances. All the software and data used in the workshop are
hosted on an Amazon Machine Image (AMI). If you want to run your
instance of the server used for this workshop, follow the directions in
the <a href="index.html#setup">Setup</a> tab.</p>
<div class="section level2">
<h2 id="citation">Citation<a class="anchor" aria-label="anchor" href="#citation"></a></h2>
<p>Please cite as: <a href="https://doi.org/10.21105/jose.00209" class="external-link"><img src="https://jose.theoj.org/papers/10.21105/jose.00209/status.svg" alt="DOI" class="figure"></a></p>
<p>Claudia Zirión Martínez; Diego Garfias Gallegos; Tania Vanessa
Arellano Fernández; Aarón Espinosa Jaime; Edder D Bustos Díaz; José Abel
Lovaco Flores; Luis Gerardo Tejero Gómez; J Abraham Avelar Rivas; Nelly
Sélem (March , 2023) A Data Carpentry- Style Metagenomics Workshop</p>
</div>
</div>



      <section id="schedule"><table class="table table schedule table-striped" role="presentation"><tbody><tr><td></td><td><a href="#setup">Setup Instructions</a></td><td> Download files required for the lesson</td>
          </tr><p></p>
      </tbody></table><p>
      The actual schedule may vary slightly depending on the topics and exercises chosen by the instructor.
      </p>
      </section><section id="setup"><section><h2 class="section-heading" id="in-a-data-carpentry-workshop">In a Data Carpentry Workshop<a class="anchor" aria-label="anchor" href="#in-a-data-carpentry-workshop"></a>
<a class="anchor" aria-label="anchor" href="#in-a-data-carpentry-workshop"></a></h2>
<hr class="half-width"><p>This workshop is designed to be run on pre-imaged Amazon Web Services
(AWS) instances (a computer with all the required programs and files to
which you will have access from your computer). Except for a spreadsheet
program and an internet browser, all of the command line software and
data used in the workshop are hosted on an Amazon Machine Image (AMI).
If you are signed up to take a Metagenomics Data Carpentry Workshop,
<strong>you do not need to worry about setting up an AMI
instance.</strong> The Carpentries staff will create an instance for
you, which will be free. This setup is accurate for both self-organized
and centrally-organized workshops. Your Instructor will provide
instructions for connecting to the AMI instance at the workshop.</p>
<p>If you are in The Carpentries-Workshop, you do not even need to
install a bash terminal; the R-studio terminal provided in the AWS-AMI
is enough to run all the commands in the lesson. Instead of connecting
by ssh, users can use the R-studio AMI terminal.</p>
<p>This lesson requires a working spreadsheet program. If you don’t have
a spreadsheet program already, you can use LibreOffice. It’s a free,
open-source spreadsheet program.</p>
</section><section><h2 class="section-heading" id="running-the-lesson-by-yourself-not-in-a-data-carpentry-workshop">Running the lesson by yourself (Not in a Data Carpentry
Workshop)<a class="anchor" aria-label="anchor" href="#running-the-lesson-by-yourself-not-in-a-data-carpentry-workshop"></a>
<a class="anchor" aria-label="anchor" href="#running-the-lesson-by-yourself-not-in-a-data-carpentry-workshop"></a></h2>
<hr class="half-width"><div class="section level3">
<h3 id="required-software">Required software<a class="anchor" aria-label="anchor" href="#required-software"></a></h3>
<p>If you are not in a Data Carpentry Workshop, the software you need is
listed in the table below. Follow the instructions in Option A
<em>or</em> Option B to have access to these programs.</p>
<table class="table"><colgroup><col width="8%"><col width="11%"><col width="3%"><col width="41%"><col width="34%"></colgroup><thead><tr class="header"><th>Software website</th>
<th>Used Version in Conda</th>
<th>Manual</th>
<th>Available for</th>
<th>Description</th>
</tr></thead><tbody><tr class="odd"><td><a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/" class="external-link">FastQC</a></td>
<td><a href="https://anaconda.org/bioconda/fastqc" class="external-link">0.11.9</a></td>
<td><a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/" class="external-link">Help</a></td>
<td>Linux, macOS, Windows</td>
<td>Quality control tool for high throughput sequence data.</td>
</tr><tr class="even"><td><a href="https://www.usadellab.org/cms/?page=trimmomatic" class="external-link">Trimmomatic</a></td>
<td><a href="https://anaconda.org/bioconda/trimmomatic" class="external-link">0.39</a></td>
<td><a href="https://github.com/usadellab/Trimmomatic" class="external-link">GitHub</a></td>
<td>Linux, macOS, Windows</td>
<td>A flexible read trimming tool for Illumina NGS data.</td>
</tr><tr class="odd"><td><a href="https://ccb.jhu.edu/software/kraken2/" class="external-link">Kraken</a></td>
<td><a href="https://anaconda.org/bioconda/kraken2" class="external-link">2.1.2</a></td>
<td><a href="https://github.com/DerrickWood/kraken2/wiki/Manual" class="external-link">GitHub</a></td>
<td>Linux, macOS</td>
<td>A tool for taxonomic assignation for reads from metagenomics</td>
</tr><tr class="even"><td><a href="https://github.com/marbl/Krona/tree/master/KronaTools" class="external-link">KronaTools</a></td>
<td><a href="https://anaconda.org/bioconda/krona" class="external-link">2.8.1</a></td>
<td><a href="https://github.com/marbl/Krona/wiki/KronaTools" class="external-link">GitHub</a></td>
<td>Linux, macOS, Windows</td>
<td>A tool for taxonomic visualization in hierarchical pie graphs.</td>
</tr><tr class="odd"><td><a href="https://sourceforge.net/projects/maxbin2/" class="external-link">MaxBin2</a></td>
<td><a href="https://anaconda.org/bioconda/maxbin2" class="external-link">2.2.7</a></td>
<td><a href="https://sourceforge.net/projects/maxbin2/" class="external-link">SourceForge</a></td>
<td>Linux, macOS</td>
<td>Tool for MAGs reconstruction</td>
</tr><tr class="even"><td><a href="https://cab.spbu.ru/software/spades/" class="external-link">Spades</a></td>
<td><a href="https://anaconda.org/bioconda/spades" class="external-link">3.15.2</a></td>
<td><a href="https://github.com/ablab/spades#sec3" class="external-link">GitHub</a></td>
<td>Linux, macOS</td>
<td>Tool for assemblies</td>
</tr><tr class="odd"><td><a href="https://github.com/smdabdoub/kraken-biom" class="external-link">Kraken-biom</a></td>
<td><a href="https://anaconda.org/bioconda/kraken-biom" class="external-link">1.2.0</a></td>
<td><a href="https://github.com/smdabdoub/kraken-biom" class="external-link">GitHub</a></td>
<td>Linux, macOS, Windows</td>
<td>Tool to convert kraken reports in R readable files</td>
</tr><tr class="even"><td><a href="https://ecogenomics.github.io/CheckM/" class="external-link">CheckM-genome</a></td>
<td><a href="https://anaconda.org/bioconda/checkm-genome" class="external-link">1.2.1</a></td>
<td><a href="https://github.com/Ecogenomics/CheckM/wiki/Quick-Start#typical-workflow" class="external-link">Wiki</a></td>
<td>Linux, macOs, Windows</td>
<td>Tool to check completeness and contamination in MAGs</td>
</tr></tbody></table></div>
<div class="section level3">
<h3 id="option-a-using-the-lessons-with-amazon-web-services-aws">Option A: Using the lessons with Amazon Web Services (AWS)<a class="anchor" aria-label="anchor" href="#option-a-using-the-lessons-with-amazon-web-services-aws"></a></h3>
<p>Follow these <a href="https://carpentries-lab.github.io/metagenomics-workshop/AMI-setup/index.html" class="external-link">instructions
on creating an Amazon instance</a>. Use the AMI
<code>ami-0f58e878fa70cc201</code> named
<code>The Carpentries Lab Metagenomics v1.0</code> listed on the
Community AMIs page. Please note that you must set your location as
<code>N. Virginia</code> to access this community AMI. You can change
your location in the upper right corner of the main AWS menu bar. The
cost of using this AMI for a few days, with the t2.medium instance type,
is very low (about USD $2.00 per user per day). Data Carpentry has
<em>no</em> control over AWS pricing structure and provides this cost
estimate without guarantees. Please read AWS documentation on pricing
for up-to-date information.</p>
<p>If you’re an Instructor or Maintainer or want to contribute to these
lessons, don’t hesitate to contact us at <a href="mailto:team@carpentries.org">team@carpentries.org</a>, and we will
start instances for you.</p>
<p>In this instance, you can use the terminal available in RStudio, and
users won’t need to install their terminals or use <code>ssh</code> (see
<a href="https://carpentries-lab.github.io/metagenomics-workshop/guide/index.html" class="external-link">Instructor
Notes</a>). <strong>If, nevertheless, you prefer that the users install
their own terminals</strong>, directions to install them are included
for each Windows, Mac OS X, and Linux below in the Option B section. For
Windows, you will need to install Git Bash, PuTTY, or the Ubuntu
Subsystem.</p>
</div>
<div class="section level3">
<h3 id="option-b-following-the-lessons-on-your-local-machine">Option B: Following the lessons on your local machine<a class="anchor" aria-label="anchor" href="#option-b-following-the-lessons-on-your-local-machine"></a></h3>
<p>If you trust that your computer is powerful enough and want to have
all the programs installed, you can follow all the workshops without
using an AWS remote machine. To do this, you will need to install all of
the software used in the workshop and obtain a copy of the dataset.
Instructions for doing this are below.</p>
<div class="section level4">
<h4 id="data"><strong>Data</strong><a class="anchor" aria-label="anchor" href="#data"></a></h4>
<p>The data used in this workshop are available on Zenodo. Please read
the Zenodo page linked below for information about the data and access
to the data files. Because this workshop works with real data, be aware
that file sizes for the data are large. <a href="https://doi.org/10.5281/zenodo.4285900" class="external-link"><img src="https://zenodo.org/badge/DOI/10.5281/zenodo.4285900.svg" alt="DOI" class="figure"></a></p>
<p>More information about these data will be presented in the <a href="https://carpentries-lab.github.io/metagenomics-analysis/01-background-metadata/index.html" class="external-link">first
episode of the Data processing and visualization for metagenomics
lesson</a>.</p>
</div>
<div class="section level4">
<h4 id="install-a-bash-terminal"><strong>Install a Bash terminal</strong><a class="anchor" aria-label="anchor" href="#install-a-bash-terminal"></a></h4>
<div id="accordionSpoiler1" class="accordion spoiler-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button spoiler-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSpoiler1" aria-expanded="false" aria-controls="collapseSpoiler1">
  <h3 class="accordion-header" id="headingSpoiler1">
  <div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="eye"></i></div>
  Windows
  </h3>
</button>
<div id="collapseSpoiler1" class="accordion-collapse collapse" aria-labelledby="headingSpoiler1" data-bs-parent="#accordionSpoiler1">
<div class="accordion-body">
<ul><li>
<p>Download the <a href="https://git-for-windows.github.io/" class="external-link">Git for
Windows installer</a>. Run the installer and follow the steps below:</p>
<ul><li>Click on “Next” four times (two times if you’ve previously installed
Git). You don’t need to change anything in the information, location,
components, and start menu screens.</li>
<li>Select “Use the nano editor by default” and click on “Next”.</li>
<li>Keep “Use Git from the Windows Command Prompt” selected and click on
“Next”. If you forget to do this, the programs that you need for the
workshop will not work properly. If this happens, rerun the installer
and select the appropriate option.</li>
<li>Select “Use bundled OpenSSH” and click on “Next”.</li>
<li>Select “Use the OpenSSL Library” and click “Next”.</li>
<li>Keep “Checkout Windows-style, commit Unix-style line endings”
selected and click on “Next”.</li>
<li>Select “Use Windows’ default console window” and click on
“Next”.</li>
<li>Select “Default (fast-forward on merge)” and click on “Next”.</li>
<li>Select “None” (Do not use a credential helper) and click on
“Next”.</li>
<li>Select “Enable file system caching” and click on “Next”.</li>
<li>Ignore “Configuring experimental options” and click on
“Install”.</li>
<li>Click on “Install”.</li>
<li>Click on “Finish”.</li>
<li>If your “HOME” environment variable is not set (or you don’t know
what this is):</li>
<li>Open command prompt (Open Start Menu, then type <code>cmd</code> and
press [Enter])</li>
<li>Type the following line into the command prompt window exactly as
shown: <code>setx HOME "%USERPROFILE%"</code>
</li>
<li>Press [Enter], and you should see
<code>SUCCESS: Specified value was saved.</code>
</li>
<li>Quit the command prompt by typing <code>exit</code> and then
pressing [Enter]</li>
<li>See the <a href="https://youtu.be/yo7Z-BEG62A" class="external-link">video tutorial</a>
for an example of how to install Git on Windows 11.</li>
</ul></li>
<li><p>An <strong>alternative option</strong> is to install PuTTY by
going to the <a href="https://www.chiark.greenend.org.uk/~sgtatham/putty/latest.html" class="external-link">the
installation page</a>. For most newer computers, click on
putty-64bit-X.XX-installer.msi to download the 64-bit version. If you
have an older laptop, you may need to get the 32-bit version
putty-X.XX-installer.msi. If you aren’t sure whether you need the 64 or
32-bit version, you can check your laptop version by following <a href="https://support.microsoft.com/en-us/help/15056/windows-32-64-bit-faq" class="external-link">the
instructions here</a>. Once the installer is downloaded, double-click on
it, and PuTTY should install.</p></li>
<li><p><strong>Another alternative option</strong> is to use the Windows
Subsystem Linux (WSL). This option is available for Windows 10 and
Windows 11 - detailed <a href="https://learn.microsoft.com/en-us/windows/wsl/install" class="external-link">instructions
are available here</a>. See the <a href="https://youtu.be/YoNdTuN-YWk" class="external-link">video tutorial</a> for an example of
how to install WSL with Ubuntu 22.04 on Windows 11.</p></li>
</ul></div>
</div>
</div>
</div>
<div id="accordionSpoiler2" class="accordion spoiler-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button spoiler-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSpoiler2" aria-expanded="false" aria-controls="collapseSpoiler2">
  <h3 class="accordion-header" id="headingSpoiler2">
  <div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="eye"></i></div>
  macOS
  </h3>
</button>
<div id="collapseSpoiler2" class="accordion-collapse collapse" aria-labelledby="headingSpoiler2" data-bs-parent="#accordionSpoiler2">
<div class="accordion-body">
<ul><li>The default shell in some versions of macOS is Bash, and Bash is
available in all versions, so no need to install anything. You access
Bash from the Terminal Application (found in /Applications/Utilities).
See how to open the terminal in the <a href="https://www.youtube.com/watch?v=FuNsWg_VzeQ" class="external-link">video tutorial</a>.
You can keep the terminal in your dock for this workshop.</li>
</ul></div>
</div>
</div>
</div>
<div id="accordionSpoiler3" class="accordion spoiler-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button spoiler-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSpoiler3" aria-expanded="false" aria-controls="collapseSpoiler3">
  <h3 class="accordion-header" id="headingSpoiler3">
  <div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="eye"></i></div>
  Linux
  </h3>
</button>
<div id="collapseSpoiler3" class="accordion-collapse collapse" aria-labelledby="headingSpoiler3" data-bs-parent="#accordionSpoiler3">
<div class="accordion-body">
<ul><li>The default shell is usually Bash, and there is usually no need to
install anything. To see if your default shell is Bash type, echo $SHELL
in a terminal and press Enter. If the message printed does not end with
<code>/bash</code>, then your default is something else, and you can run
Bash by typing <code>bash</code>.</li>
</ul></div>
</div>
</div>
</div>
</div>
<div class="section level4">
<h4 id="install-miniconda3"><strong>Install Miniconda3</strong><a class="anchor" aria-label="anchor" href="#install-miniconda3"></a></h4>
<p>These instructions assume familiarity with the command line and with
installation in general. There are different operating systems and many
different versions of operating systems and environments, so these may
not work on your computer. If an installation doesn’t work for you,
please refer to the user guide for the tool listed in the table above.
If you have difficulties with the installations or find better ways to
install things in your operating system, please raise an <a href="https://github.com/carpentries-lab/metagenomics-analysis/issues" class="external-link">Issue</a>
to let us know.</p>
<p>To make a <a href="https://conda.io/projects/conda/en/latest/index.html" class="external-link">Conda</a>
environment, first, you need to install Conda. We recommend installing
the <a href="https://docs.conda.io/en/latest/miniconda.html" class="external-link">Miniconda3</a>
version. Miniconda is a package manager that includes Conda and its
dependencies and simplifies the installation process. Please first
install Miniconda3 (installation instructions below) and then proceed to
the installation of the environment.</p>
<div id="accordionSpoiler4" class="accordion spoiler-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button spoiler-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSpoiler4" aria-expanded="false" aria-controls="collapseSpoiler4">
  <h3 class="accordion-header" id="headingSpoiler4">
  <div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="eye"></i></div>
  Linux
  </h3>
</button>
<div id="collapseSpoiler4" class="accordion-collapse collapse" aria-labelledby="headingSpoiler4" data-bs-parent="#accordionSpoiler4">
<div class="accordion-body">
<p>To install miniconda3, see the <a href="https://youtu.be/0PqwShSDH20" class="external-link">video tutorial</a></p>
</div>
</div>
</div>
</div>
<div id="accordionSpoiler5" class="accordion spoiler-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button spoiler-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSpoiler5" aria-expanded="false" aria-controls="collapseSpoiler5">
  <h3 class="accordion-header" id="headingSpoiler5">
  <div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="eye"></i></div>
  MacOSX
  </h3>
</button>
<div id="collapseSpoiler5" class="accordion-collapse collapse" aria-labelledby="headingSpoiler5" data-bs-parent="#accordionSpoiler5">
<div class="accordion-body">
<p>In a terminal type:</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="ex">$</span> curl <span class="at">-O</span> https://repo.anaconda.com/miniconda/Miniconda3-latest-MacOSX-x86_64.sh</span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a><span class="ex">$</span> bash Miniconda3-latest-MacOSX-x86_64.sh</span></code></pre>
</div>
<p>Then, follow the instructions that you are prompted with on the
screen to install Miniconda3.</p>
</div>
</div>
</div>
</div>
<div id="accordionSpoiler6" class="accordion spoiler-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button spoiler-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSpoiler6" aria-expanded="false" aria-controls="collapseSpoiler6">
  <h3 class="accordion-header" id="headingSpoiler6">
  <div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="eye"></i></div>
  WSL
  </h3>
</button>
<div id="collapseSpoiler6" class="accordion-collapse collapse" aria-labelledby="headingSpoiler6" data-bs-parent="#accordionSpoiler6">
<div class="accordion-body">
<p>See the video tutorial, <a href="https://youtu.be/owQgZoE-GrY" class="external-link">installing Miniconda3 on WSL
Ubuntu</a></p>
</div>
</div>
</div>
</div>
</div>
<div class="section level4">
<h4 id="install-the-metagenomics-environment"><strong>Install the metagenomics environment</strong><a class="anchor" aria-label="anchor" href="#install-the-metagenomics-environment"></a></h4>
<p>Once your Miniconda3 is ready, follow these instructions to install
and activate the metagenomics environment.</p>
<div id="accordionSpoiler7" class="accordion spoiler-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button spoiler-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSpoiler7" aria-expanded="false" aria-controls="collapseSpoiler7">
  <h3 class="accordion-header" id="headingSpoiler7">
  <div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="eye"></i></div>
  Linux: Option 1 (recommended)
  </h3>
</button>
<div id="collapseSpoiler7" class="accordion-collapse collapse" aria-labelledby="headingSpoiler7" data-bs-parent="#accordionSpoiler7">
<div class="accordion-body">
<p>The easier way to install the environment is using the <a href="https://github.com/carpentries-lab/metagenomics-analysis/blob/gh-pages/files/spec-file-Ubuntu22.txt" class="external-link">specifications
file</a> for Linux Ubuntu 22.04, which has the exact versions of each
tool in this environment. You can use the spec file as follows:</p>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="ex">$</span> conda create <span class="at">--name</span> metagenomics <span class="at">--file</span> spec-file-Ubuntu22.txt</span></code></pre>
</div>
<p>More information about how to use environments and spec files is
available at <a href="https://conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html" class="external-link">conda
documentation</a></p>
</div>
</div>
</div>
</div>
<div id="accordionSpoiler8" class="accordion spoiler-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button spoiler-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSpoiler8" aria-expanded="false" aria-controls="collapseSpoiler8">
  <h3 class="accordion-header" id="headingSpoiler8">
  <div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="eye"></i></div>
  Linux: option 2
  </h3>
</button>
<div id="collapseSpoiler8" class="accordion-collapse collapse" aria-labelledby="headingSpoiler8" data-bs-parent="#accordionSpoiler8">
<div class="accordion-body">
<p>Another way to create an environment is with a <code>ỳml</code> file.
This environment can be modified by adding or deleting tools in a file
<code>metagenomics-Ubuntu22.yml</code>.</p>
<p>In Ubuntu 22.04, copy this file <a href="https://github.com/carpentries-lab/metagenomics-analysis/blob/gh-pages/files/metagenomics-Ubuntu22.yml" class="external-link">metagenomics-Ubuntu22.yml</a>
to your computer and follow the instructions in the <a href="https://youtu.be/0Ks1k9gg3E0" class="external-link">video tutorial</a></p>
</div>
</div>
</div>
</div>
<div id="accordionSpoiler9" class="accordion spoiler-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button spoiler-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSpoiler9" aria-expanded="false" aria-controls="collapseSpoiler9">
  <h3 class="accordion-header" id="headingSpoiler9">
  <div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="eye"></i></div>
  MacOSX
  </h3>
</button>
<div id="collapseSpoiler9" class="accordion-collapse collapse" aria-labelledby="headingSpoiler9" data-bs-parent="#accordionSpoiler9">
<div class="accordion-body">
<p>It has been difficult to find compatibility between all the
dependencies of each package installed in the metagenomics environment.
In the case of the latest version of macOS (Monterey), the MaxBin2
package can be installed, but it does not fully work at the time of use.
Copy the file <a href="https://github.com/carpentries-lab/metagenomics-analysis/blob/gh-pages/files/metagenomics-macOS.yml" class="external-link">metagenomics-macOS.yml</a>
in your computer and run:</p>
<div class="codewrapper sourceCode" id="cb3">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="ex">$</span> conda env create <span class="at">-f</span> metagenomics-macOS.yml</span></code></pre>
</div>
</div>
</div>
</div>
</div>
<div id="accordionSpoiler10" class="accordion spoiler-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button spoiler-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSpoiler10" aria-expanded="false" aria-controls="collapseSpoiler10">
  <h3 class="accordion-header" id="headingSpoiler10">
  <div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="eye"></i></div>
  WSL
  </h3>
</button>
<div id="collapseSpoiler10" class="accordion-collapse collapse" aria-labelledby="headingSpoiler10" data-bs-parent="#accordionSpoiler10">
<div class="accordion-body">
<p>In the case of Windows Subsystem for Linux WSL Ubuntu 22.04, the
MaxBin2 package has an incompatibility with the checkm-genome package,
so we have decided to leave it out of the metagenomics environment and
create its own environment with the file(<a href="https://github.com/carpentries-lab/metagenomics-analysis/blob/gh-pages/files/metagenomics-maxbin.yml" class="external-link">metagenomics-maxbin.yml</a>).
The file for the metagenomics environment is <a href="https://github.com/carpentries-lab/metagenomics-analysis/blob/gh-pages/files/metagenomics-WSLUbuntu.yml" class="external-link">metagenomics-WSLUbuntu.yml</a>
See the <a href="https://youtu.be/kHYmffZYxcI" class="external-link">video tutorial</a></p>
<div class="codewrapper sourceCode" id="cb4">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a><span class="ex">$</span> conda env create <span class="at">-f</span> metagenomics-maxbin.yml</span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a><span class="ex">$</span> conda env create <span class="at">-f</span> metagenomics-WSLUbuntu.yml</span></code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="section level4">
<h4 id="execute-some-remaining-installation-scripts"><strong>Execute some remaining installation scripts</strong><a class="anchor" aria-label="anchor" href="#execute-some-remaining-installation-scripts"></a></h4>
<p>Change <code>dcuser</code> with your own username. And run all these
lines:</p>
<div class="codewrapper sourceCode" id="cb5">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a><span class="fu">bash</span> /home/dcuser/.miniconda3/envs/metagenomics/opt/krona/updateTaxonomy.sh                                </span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a><span class="fu">wget</span> ftp://ftp.ncbi.nih.gov/pub/taxonomy/taxdump.tar.gz </span>
<span id="cb5-3"><a href="#cb5-3" tabindex="-1"></a><span class="fu">tar</span> <span class="at">-xzf</span> taxdump.tar.gz </span>
<span id="cb5-4"><a href="#cb5-4" tabindex="-1"></a><span class="fu">mkdir</span> .taxonkit</span>
<span id="cb5-5"><a href="#cb5-5" tabindex="-1"></a><span class="fu">cp</span> names.dmp nodes.dmp delnodes.dmp merged.dmp /home/dcuser/.taxonkit</span>
<span id="cb5-6"><a href="#cb5-6" tabindex="-1"></a><span class="fu">rm</span> <span class="pp">*</span>dmp readme.txt taxdump.tar.gz gc.prt </span></code></pre>
</div>
</div>
<div class="section level4">
<h4 id="install-r-and-rstudio"><strong>Install R and RStudio</strong><a class="anchor" aria-label="anchor" href="#install-r-and-rstudio"></a></h4>
<p>R and RStudio are two separate pieces of software:</p>
<p>R is a programming language that is especially powerful for data
exploration, visualization, and statistical analysis RStudio is an
integrated development environment (IDE) that makes using R easier. In
this course, we use RStudio to interact with R.</p>
<div id="accordionSpoiler11" class="accordion spoiler-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button spoiler-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSpoiler11" aria-expanded="false" aria-controls="collapseSpoiler11">
  <h3 class="accordion-header" id="headingSpoiler11">
  <div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="eye"></i></div>
  Mac OS X
  </h3>
</button>
<div id="collapseSpoiler11" class="accordion-collapse collapse" aria-labelledby="headingSpoiler11" data-bs-parent="#accordionSpoiler11">
<div class="accordion-body">
<ul><li><p>Download R from the <a href="https://cran.r-project.org/bin/macosx/" class="external-link">CRAN website</a>.</p></li>
<li><p>Select the .pkg file for the latest R version</p></li>
<li><p>Double-click on the downloaded file to install R</p></li>
<li><p>It is also a good idea to install <a href="https://www.xquartz.org/" class="external-link">XQuartz</a> (needed by some
packages)</p></li>
<li><p>Go to the <a href="https://www.rstudio.com/products/rstudio/download/#download" class="external-link">RStudio
download</a> page</p></li>
<li><p>Under Installers, select RStudio x.yy.zzz - Mac OS X 10.6+
(64-bit) (where x, y, and z represent version numbers)</p></li>
<li><p>Double-click the file to install RStudio</p></li>
<li><p>Once it’s installed, open RStudio to ensure it works and you
don’t get any error messages.</p></li>
</ul></div>
</div>
</div>
</div>
<div id="accordionSpoiler12" class="accordion spoiler-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button spoiler-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSpoiler12" aria-expanded="false" aria-controls="collapseSpoiler12">
  <h3 class="accordion-header" id="headingSpoiler12">
  <div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="eye"></i></div>
  Windows
  </h3>
</button>
<div id="collapseSpoiler12" class="accordion-collapse collapse" aria-labelledby="headingSpoiler12" data-bs-parent="#accordionSpoiler12">
<div class="accordion-body">
<ul><li><p>Download R from the <a href="https://cran.r-project.org/bin/windows/base/" class="external-link">CRAN
website</a>.</p></li>
<li><p>Run the .exe file that was just downloaded</p></li>
<li><p>Go to the <a href="https://www.rstudio.com/products/rstudio/download/#download" class="external-link">RStudio
download page</a></p></li>
<li><p>Under Installers select RStudio x.yy.zzz - Windows Vista/7/8/10
(where x, y, and z represent version numbers)</p></li>
<li><p>Double-click the file to install it</p></li>
<li><p>Once it’s installed, open RStudio to ensure it works and you
don’t get any error messages.</p></li>
</ul></div>
</div>
</div>
</div>
<div id="accordionSpoiler13" class="accordion spoiler-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button spoiler-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSpoiler13" aria-expanded="false" aria-controls="collapseSpoiler13">
  <h3 class="accordion-header" id="headingSpoiler13">
  <div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="eye"></i></div>
  Linux
  </h3>
</button>
<div id="collapseSpoiler13" class="accordion-collapse collapse" aria-labelledby="headingSpoiler13" data-bs-parent="#accordionSpoiler13">
<div class="accordion-body">
<ul><li><p>Follow the instructions for your distribution from <a href="https://cloud.r-project.org/bin/linux" class="external-link">CRAN</a>. They provide
information to get the most recent version of R for common
distributions. For most distributions, you could use your package
manager (e.g., for Debian/Ubuntu, run sudo apt-get install r-base, and
for Fedora, sudo yum install R). However, we don’t recommend this
approach as the versions provided by this are usually out of date. In
any case, make sure you have at least R 3.3.1.</p></li>
<li><p>Go to the <a href="https://www.rstudio.com/products/rstudio/download/#download" class="external-link">RStudio</a>
download page</p></li>
<li><p>Under Installers, select the version that matches your
distribution, and install it with your preferred method (e.g., with
Debian/Ubuntu ´sudo dpkg -i rstudio-x.yy.zzz-amd64.deb´ at the
terminal).</p></li>
<li><p>Once it’s installed, open RStudio to ensure it works and you
don’t get any error messages.</p></li>
</ul></div>
</div>
</div>
</div>
</div>
<div class="section level4">
<h4 id="install-r-libraries"><strong>Install R libraries</strong><a class="anchor" aria-label="anchor" href="#install-r-libraries"></a></h4>
<table class="table"><colgroup><col width="13%"><col width="17%"><col width="5%"><col width="63%"></colgroup><thead><tr class="header"><th>Software</th>
<th>Version</th>
<th>Manual</th>
<th>Description</th>
</tr></thead><tbody><tr class="odd"><td><a href="https://github.com/joey711/phyloseq" class="external-link">phyloseq</a></td>
<td>1.39.1</td>
<td><a href="https://github.com/joey711/phyloseq" class="external-link">GitHub</a></td>
<td>Explore, manipulate and analyze microbiome profiles with R</td>
</tr><tr class="even"><td><a href="https://ggplot2.tidyverse.org/" class="external-link">ggplot2</a></td>
<td>3.3.6</td>
<td><a href="https://github.com/tidyverse/ggplot2" class="external-link">GitHub</a></td>
<td>System for declaratively creating graphics, based on The Grammar of
Graphics</td>
</tr></tbody></table><p>Type these commands in your console:</p>
<div class="codewrapper sourceCode" id="cb6">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode r" tabindex="0"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a><span class="sc">&gt;</span> <span class="fu">install.packages</span>(<span class="st">"phyloseq"</span>)</span>
<span id="cb6-2"><a href="#cb6-2" tabindex="-1"></a><span class="sc">&gt;</span> <span class="fu">install.packages</span>(<span class="st">"ggplot2"</span>)</span></code></pre>
</div>
</div>
</div>
</section></section></div>
<!-- END  : inst/pkgdown/templates/content-syllabus.html -->

      </main></div><!--/div.row-->
      		<footer class="row footer mx-md-3"><hr><div class="col-md-6">
        <p>This lesson is subject to the <a href="CODE_OF_CONDUCT.html">Code of Conduct</a></p>
        <p>

        <a href="https://github.com/carpentries-lab/metagenomics-workshop/edit/main/index.md" class="external-link">Edit on GitHub</a>

	
        | <a href="https://github.com/carpentries-lab/metagenomics-workshop/blob/main/CONTRIBUTING.md" class="external-link">Contributing</a>
        | <a href="https://github.com/carpentries-lab/metagenomics-workshop/" class="external-link">Source</a></p>
				<p><a href="https://github.com/carpentries-lab/metagenomics-workshop/blob/main/CITATION.cff" class="external-link">Cite</a> | <a href="mailto:nselem@matmor.unam.mx">Contact</a> | <a href="https://carpentries.org/about/" class="external-link">About</a></p>
			</div>
			<div class="col-md-6">

        <p>Materials licensed under <a href="LICENSE.html">CC-BY 4.0</a> by the authors</p>

        <p>Template licensed under <a href="https://creativecommons.org/licenses/by-sa/4.0/" class="external-link">CC-BY 4.0</a> by <a href="https://carpentries.org/" class="external-link">The Carpentries</a></p>
        <p>Built with <a href="https://github.com/carpentries/sandpaper/tree/0.16.5" class="external-link">sandpaper (0.16.5)</a>, <a href="https://github.com/carpentries/pegboard/tree/0.7.6" class="external-link">pegboard (0.7.6)</a>, and <a href="https://github.com/carpentries/varnish/tree/1.0.3" class="external-link">varnish (1.0.3)</a></p>
			</div>
		</footer></div> <!-- / div.container -->
	<div id="to-top">
		<a href="#top">
      <i class="search-icon" data-feather="arrow-up" role="img" aria-label="Back To Top"></i><br><!-- <span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top --><span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top
		</a>
	</div>
  <script type="application/ld+json">
    {
  "@context": "https://schema.org",
  "@type": "TrainingMaterial",
  "@id": "https://carpentries-lab.github.io/metagenomics-workshop/instructor/index.html",
  "inLanguage": "en",
  "dct:conformsTo": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
  "description": "A Carpentries Lesson teaching foundational data and coding skills to researchers worldwide",
  "keywords": "software, data, lesson, The Carpentries",
  "name": "Metagenomics Workshop Overview",
  "creativeWorkStatus": "active",
  "url": "https://carpentries-lab.github.io/metagenomics-workshop/instructor/index.html",
  "identifier": "https://carpentries-lab.github.io/metagenomics-workshop/instructor/index.html",
  "dateCreated": "2021-05-27",
  "dateModified": "2024-07-16",
  "datePublished": "2024-07-16"
}

  </script><script>
		feather.replace();
	</script><!-- Matomo --><script>
          var _paq = window._paq = window._paq || [];
          /* tracker methods like "setCustomDimension" should be called before "trackPageView" */
          _paq.push(["setDocumentTitle", document.domain + "/" + document.title]);
          _paq.push(["setDomains", ["*.lessons.carpentries.org","*.datacarpentry.github.io","*.datacarpentry.org","*.librarycarpentry.github.io","*.librarycarpentry.org","*.swcarpentry.github.io", "*.carpentries.github.io"]]);
          _paq.push(["setDoNotTrack", true]);
          _paq.push(["disableCookies"]);
          _paq.push(["trackPageView"]);
          _paq.push(["enableLinkTracking"]);
          (function() {
              var u="https://matomo.carpentries.org/";
              _paq.push(["setTrackerUrl", u+"matomo.php"]);
              _paq.push(["setSiteId", "1"]);
              var d=document, g=d.createElement("script"), s=d.getElementsByTagName("script")[0];
              g.async=true; g.src="https://matomo.carpentries.org/matomo.js"; s.parentNode.insertBefore(g,s);
          })();
        </script><!-- End Matomo Code --></div></body></html><!-- END:   inst/pkgdown/templates/layout.html-->

